### PR TITLE
Enable universal wheel build on CI

### LINKED ISSUE #881: publish purepython/universal wheel?
## Long story short

CD pipeline does not publish a pure python version on PyPI.

I don't have a pressing need for this but we're investigating whether we can make some sigstore-python functionality available as a pure python module and this dependency came up as one of the issues preventing that.

## Expected behaviour

Since multidict can be built as a pure Python module I expected a wheel to be available in PyPI as well

## Actual behaviour

cd-cd.yml does not seem to build a purepython wheel, but the ability is there in the build system itself.

## Details

I can provide a PR  but I admit I haven't tested it (CD pipelines being a bit tricky to test). The changes needed seem to be:
* add a new build matrix variable `noextensions` (default 0) , and add a new build into the build matrix that uses noextensions: 1
* based on a matrix variable set MULTIDICT_NO_EXTENSIONS during the build
* fix bug in MULTIDICT_NO_EXTENSIONS handling so that "MULTIDICT_NO_EXTENSIONS=0" actually uses extensions

### PR DESCRIPTION
**Pull request by jku**
## What do these changes do?

Enable building of universal (pure python) build on the CD release pipeline.

## Are there changes in behavior for the user?

A universal wheel will be available on PyPI.

## Related issue number

Fixes #881

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] Add a new news fragment into the `CHANGES` folder

## Details

The PR has not been tested yet: running the workflow in a fork does not work so I'll have to rely on the test results here in the PR...

### CODE REVIEW

**Comment by jku** [comment]
```
[2023-09-14T13:15:25.468Z] ['info'] Pinging Codecov: https://codecov.io/upload/v4?package=github-action-2.1.0-uploader-0.6.2&token=*******&branch=enable-universal-wheel&build=6182260616&build_url=https%3A%2F%2Fgithub.com%2Faio-libs%2Fmultidict%2Factions%2Fruns%2F6182260616&commit=ec77ff325aa74f62399ac5d0af7c592e493c5953&job=CI&pr=883&service=github-actions&slug=aio-libs%2Fmultidict&name=&tag=&flags=&parent=
[2023-09-14T13:15:25.730Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
Error: Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v2.1.0/dist/codecov' failed with exit code 255
```

aio-libs/upload-coverage fails like that. I don't think this is related to my changes.

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -166,8 +166,11 @@ jobs:
       matrix:
         os: [ubuntu, windows, macos]
         qemu: ['']
+        noextensions: ['0']
         include:
-          # Split ubuntu job for the sake of speed-up
+          # Add a pure python build and qemu builds
+        - os: ubuntu
+          noextensions: '1'
```
Comment (webknjaz) [inline .github/workflows/ci-cd.yml:173]: This is not the right places to add it. It would essentially run the same wheel creation under each Python version + waste a CI job. It can be done together with the sdist. Although, that should also be updated as it still invokes `setup.py`.

I'm doing a round of maintenance updates across the repos, including packaging and CI changes. frozenlist and yarl are updated, multidict is next on my list. So I'll take care of this in a standardized way.

File: CHANGES.rst [hunk 2]
```diff
@@ -7,7 +7,7 @@ Changelog
     file is managed by towncrier. You *may* edit previous change logs to
     fix problems like typo corrections or such.
     To add a new change log entry, please see
-    https://pip.pypa.io/en/latest/development/#adding-a-news-entry
+    https://pip.pypa.io/en/latest/development/contributing/#news-entries
```
Comment (webknjaz) [inline CHANGES.rst:10]: This update could be useful as a separate PR.

File: CHANGES/881.feature.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Enable publishing a universal (pure Python) wheel on PyPI
```
Comment (webknjaz) [inline CHANGES/881.feature.rst:1]: The change notes should use the past tense with the end-users as the target audience in mind. They should include full sentences with periods et al.
Also, "universal" is an incorrect term — conventionally, wheels tagged as `py2.py3` used to be called universal. But we should only have `py3` in the tag.

File: multidict/_compat.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import os
 import platform
 
-NO_EXTENSIONS = bool(os.environ.get("MULTIDICT_NO_EXTENSIONS"))
+NO_EXTENSIONS = os.environ.get("MULTIDICT_NO_EXTENSIONS", "0") == "1"
```
Comment (webknjaz) [inline multidict/_compat.py:4]: This is a semantic change that isn't even mentioned in the changelog. It's undesired at this point. But if we were to change it, this would mean a feature PR, made with more care.

File: setup.py [hunk 5]
```diff
@@ -6,7 +6,7 @@
 
 from setuptools import Extension, setup
 
-NO_EXTENSIONS = bool(os.environ.get("MULTIDICT_NO_EXTENSIONS"))
+NO_EXTENSIONS = os.environ.get("MULTIDICT_NO_EXTENSIONS", "0") == "1"
```
Comment (webknjaz) [inline setup.py:9]: Same as the previous comment. Although, it doesn't matter much now, since I'll align the packaging with frozenlist and yarl soon.

**Comment by webknjaz** [comment]
Implemented in https://github.com/aio-libs/multidict/commit/7ba0e7205e4e5c74abd7e8f352e563427eed2501.
